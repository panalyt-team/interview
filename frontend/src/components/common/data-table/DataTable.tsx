import React from "react";
import { CheckboxGroupProperties, DataTableRowValue } from "../interfaces";
import DataTableHeader from "./data-table-header/DataTableHeader";
import DataTableRow from "./data-table-row/DataTableRow";

export interface DataTableProps {
    columnsNames: string[];
    rowValues: DataTableRowValue[][];
    filters: CheckboxGroupProperties;
}

export const DataTable = ({ columnsNames, rowValues, filters }: DataTableProps): JSX.Element => {

    return (
        <>
            <DataTableHeader columns={columnsNames} filters={filters}/>
            <div>
                {
                    rowValues.map((value: DataTableRowValue[], index: number) => 
                        <DataTableRow key={index} rowValues={value}/>
                    )
                }             
            </div>
        </>
    )
};

export default DataTable;