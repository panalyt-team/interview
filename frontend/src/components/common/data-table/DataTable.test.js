import { shallow } from 'enzyme';
import DataTable from './DataTable';

describe("DataTable", () => {

    const filters = {
        groupLabel:"Countries",
        checkboxes:[{"checked":true,"label":"All countries"},{"checked":false,"label":"China"},{"checked":false,"label":"Japan"}],
        changeHandler: () => {}
    };
    
    const rowValues = [
        ["China", "8 516,87 $", { "label": "+9.12%", "color": "#adff2e" }],
        ["Japan", "8 305,03 $", { "label": "+24.36%", "color": "#adff2e" }],
        ["Philippines", "8 215,59 $", { "label": "+17.09%", "color": "#adff2e" }]
    ];

    it("renders correctly", () => {
        const wrapper = shallow(<DataTable columns={["col1", "col2", "col3"]} filters={filters} rowValues={rowValues}/>);

        expect(wrapper).toMatchSnapshot();
    });

});