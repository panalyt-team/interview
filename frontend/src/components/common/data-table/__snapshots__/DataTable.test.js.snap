// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable renders correctly 1`] = `ShallowWrapper {}`;
